In [4]:
import iamend_ci as ci

In [9]:
exp=ci.exp('./iamend_ci/datos/marzo2023/m1')

                  archivo  conductividad  espesor bobina  muestras
0      Exp_m1_347-M02.csv     1118491.27  0.02340     m1   347-M02
1  Exp_m1_1010-M01_m1.csv     3830000.00  0.06500     m1  1010-M01
2         Exp_m1_aire.csv           0.00  0.00000     m1      aire
3         Exp_m1_P057.csv     3948000.00  0.14957     m1      P057


In [10]:
exp.fitmues()
exp.info

Ajustando z1 effectivo
z1 = 0.9690129800536345 mm


archivo  conductividad  espesor bobina  muestras     mueff  \
0      Exp_m1_347-M02.csv     1118491.27  0.02340     m1   347-M02  43.04731   
1  Exp_m1_1010-M01_m1.csv     3830000.00  0.06500     m1  1010-M01  65.45817   
2         Exp_m1_aire.csv           0.00  0.00000     m1      aire       NaN   
3         Exp_m1_P057.csv     3948000.00  0.14957     m1      P057       NaN   

         R2  
0  0.987635  
1  0.985920  
2       NaN  
3       NaN

In [11]:
exp2=ci.exp('./iamend_ci/datos/marzo2023/pp1')
exp2.fitmues()
exp2.info

                archivo  conductividad  espesor bobina  muestras
0  Exp_pp1_1010-M01.csv     3830000.00  0.06500    pp1  1010-M01
1      Exp_pp1_aire.csv           0.00  0.00000    pp1      aire
2   Exp_pp1_347-M02.csv     1118491.27  0.02340    pp1   347-M02
3      Exp_pp1_P057.csv     3948000.00  0.14957    pp1      P057
Ajustando z1 effectivo
z1 = 1.4782830446383313 mm


archivo  conductividad  espesor bobina  muestras      mueff  \
0  Exp_pp1_1010-M01.csv     3830000.00  0.06500    pp1  1010-M01  70.986921   
1      Exp_pp1_aire.csv           0.00  0.00000    pp1      aire        NaN   
2   Exp_pp1_347-M02.csv     1118491.27  0.02340    pp1   347-M02  46.857815   
3      Exp_pp1_P057.csv     3948000.00  0.14957    pp1      P057        NaN   

         R2  
0  0.984197  
1       NaN  
2  0.974511  
3       NaN

# Intervalo de confianza para parametros de un ajuste no lineal

En orden de poder comparar ambas mediciones y tener un metodo robusto para la caracterizacion de la permeabilidad, necesitamos definir un intervalo de confianza para el parametro $\mu$

La estimacion de $\mu$ se realiza con la funcion `optimize.curve_fit` de la libreria `scipy` una implementacion de  non-linear least squares para ajustar una funcion $f$ (modelo analitico) a los datos de impedancia normalizados.  el algorithm uses the Levenberg-Marquardt algorithm through leastsq.

https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.curve_fit.html

https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.least_squares.html#scipy.optimize.least_squares


Segun la libreria podemos definir el error en $\mu$ como `np.sqrt(np.diag(pcov))`.

```
pcov2-D array
The estimated covariance of popt. The diagonals provide the variance of the parameter estimate. To compute one standard deviation errors on the parameters use perr = np.sqrt(np.diag(pcov)).

How the sigma parameter affects the estimated covariance depends on absolute_sigma argument, as described above.

If the Jacobian matrix at the solution doesn’t have a full rank, then ‘lm’ method returns a matrix filled with np.inf, on the other hand ‘trf’ and ‘dogbox’ methods use Moore-Penrose pseudoinverse to compute the covariance matrix.

```
### Montecarlo

https://cran.r-project.org/web/packages/nlraa/vignettes/Confidence-Bands.html

## otros

https://stat.ethz.ch/~stahel/courses/cheming/nlreg10E.pdf

### Bootstrap resampling


- Let $X$ denote your training dataset. Let $n$ denote the number of samples in your training data. Let $k$ denote the number of resampling iterations you want to perform. The more the better, but k should probably be no fewer than 1000.

- for $i=1,2,…k$ take a random sample $X_i$ (with replacement) of size $n$ from $X$. Train your model and calculate your model paramters. Let $\mu_i$ denote your fitted parameters trained on the ith resampled data set.

- You can now calculate confidence intervals by determining the quantiles of $\hat{\mu} =[ \mu_0,\mu_1,\mu_2,...\mu_k]$ 

For example, to obtain a 95% confidence interval, calculate the 2.5% and 97.5% quantiles of $\hat{\mu}$.

    [1]Reference: Chapters seven and eight of 'The Elements of Statistical Learning'